src/OneExplorer.ts

```diff
@@ -96,41 +117,38 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
   }
 
   private getTree(rootPath: vscode.Uri): Node {
-    const node: Node = {
-      type: NodeType.directory,
-      name: path.parse(rootPath.fsPath).base,
-      childNodes: [],
-      uri: rootPath
-    };
-    this.searchNode(node, path.dirname(rootPath.fsPath));
+    const node = new Node(NodeType.directory, [], rootPath);
+
+    this.searchNode(node);
     return node;
   }
 
-  private searchNode(node: Node, dirPath: string) {
-    const dirpath = path.join(dirPath, node.name);
-    const files = fs.readdirSync(dirpath);
+  /**
+   * Construct a tree under the given node
+   * @returns void
+   */
+  private searchNode(node: Node) {
+    const files = fs.readdirSync(node.path);
 
-    for (const fn of files) {
-      const fpath = path.join(dirpath, fn);
+    for (const fname of files) {
+      const fpath = path.join(node.path, fname);
       const fstat = fs.statSync(fpath);
 
       if (fstat.isDirectory()) {
-        const dirNode: Node =
-            {type: NodeType.directory, name: fn, childNodes: [], uri: vscode.Uri.file(fpath)};
+        const childNode = new Node(NodeType.directory, [], vscode.Uri.file(fpath));
```

Comment:
It kinda unrelated changes like renaming, but please assume this as a part of cleanup. As they say a boy scout of code cleaning-  'Always leave the code better than you found it.'